v2/pkg/operators/common/dsl/dsl.go

```diff
@@ -31,109 +31,192 @@ const (
 	withMaxRandArgsSize  = withCutSetArgsSize
 )
 
+var ErrDSLArguments = errors.New("invalid arguments provided to dsl")
```

Comment:
We should be more specific in the error message like including the method signature with self-explanatory variable names and variable types
e.g.
"invalid arguments given. The method signature is `toupper(value string) string`"

Reply:
The idea to include the validations was good though!

Reply:
DSL functions don't have access to the function name at runtime, so we'll need to figure out a way to get the name of the called function. We can leave this task to another issue, I'll keep the comment open.